src/measure/OSRunner.cpp

```diff
@@ -784,6 +786,68 @@ namespace measure {
     return result;
   }
 
+  Json::Value OSRunner::getArgumentValues(std::vector<OSArgument>& script_arguments, const std::map<std::string, OSArgument>& user_arguments) {
+    Json::Value argument_values;
+    std::string name;
+    for (const OSArgument& script_argument : script_arguments) {
+      name = script_argument.name();
+      auto it = user_arguments.find(name);
+      if (it != user_arguments.end()) {
+        Json::Value root = it->second.toJSON();
+        if (auto value = root["value"]) {
+          argument_values[name] = value;
+        }
+      }
+    }
+    return argument_values;
+  }
+
+  Json::Value OSRunner::getPastStepValuesForMeasure(const std::string& measure_name) {
```

Comment:
* I'm still confused about the difference between the getPastStepValuesForMeasure and getPastStepValuesForName, but I'll figure that out.
* The above is mostyl due to how unreadable both methods are. The flow logic places an unnecessarily high cognitive burden on the reader. I'll revamp   
* Both of these should be marked const.
    * Generally speaking, the entire existing OSRunner class is completely messed up as far as const-correctness and staticness of methods that can be (like getOptionalXXXArgument), but I'm wary of changing this since it's an ABI break